content/blog/2023-12-11-circ-bitwidth-adjustment

```diff
@@ -0,0 +1,115 @@
++++
+title = "CirC Bitwidth Adjustment Implementation"
+[extra]
+bio = """
+  Collin Zhang and Melinda Fang are students at Cornell
+"""
+[[extra.authors]]
+name = "Collin Zhang"
+link = "https://collinzrj.github.io/"  # Links are optional.
+[[extra.authors]]
+name = "Melinda Fang"
++++
+
+## Link to the project
+We are currently working on creating a PR to include the change to the project. At this point, the best way to see the changes we have made is via this link which compares the latest commit with the commit before our changes https://github.com/collinzrj/circ/compare/a26533baadfd674f7d7d766b995a9a78a02d4ebd..61fb7c0ee2946cf84b92d962236ffe9533cb6c53
+
+We mainly make changes to these files
+```
+## The new optimization pass is implemented in these two files
+src/ir/opt/mod.rs
+src/ir/opt/short_int_adj.rs
+## The logic to handle non-adjust operations in the constraint generation system
+src/target/r1cs/trans.rs
+```
```

Comment:
Recall that your audience for this blog post is someone who is interested in compilers and your general topic but is not intimately familiar with the code of the CirC project. These filenames are unlikely to be meaningful to the readers. There is no need to include that level of code-level detail.